### PR TITLE
E2E framework changes for context tooling and test cases

### PR DESCRIPTION
**Pull request by chandrareddyp**
### What this PR does / why we need it
This PR has changes to E2E test framework for `context` command, it has below updates:
- Tooling and test cases for context command (k8s and tmc targets)
- GitHub workflow update to skip tmc tests if secret variables not
  set
- adding kind and kubectl tools, and build code and create bin/tanzu binary within the project folder
- `linter` has enabled for the e2e test code and e2e framework code 

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
- `make e2e-cli-core` has been tested with tmc use cases (when environment variables are set) and without tmc use cases (when environment variables are not set)
- GitHub workflow actions are successful for the e2e test cases execution

### Release note

```release-note

```

### CODE REVIEW

File: test/e2e/framework/context_create_operations.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// ContextCreateOps helps to run context create command
+type ContextCreateOps interface {
+	// CreateConextWithEndPoint creates a context with a given endpoint URL
+	CreateConextWithEndPoint(contextName, endpoint string) error
+	// CreateConextWithEndPoint creates a context with a given endpoint URL
```
Comment (anujc25) [inline test/e2e/framework/context_create_operations.go:16]:
```suggestion
	// CreateConextWithEndPointStaging creates a context with a given endpoint URL for staging
```

File: test/e2e/framework/context_create_operations.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	CreateConextWithEndPointStaging(contextName, endpoint string) error
```
Comment (anujc25) [inline test/e2e/framework/context_create_operations.go:17]: Just for understanding: Why do we need `--staging` when creating context with `--endpoint`?

File: test/e2e/framework/framework.go [hunk 3]
```diff
@@ -71,7 +100,14 @@ func init() {
...
 	pwd, _ := os.Getwd()
-	os.Setenv("PATH", pwd+HackToolsBinDir+":"+pwd+TanzuBinDir+":"+os.Getenv("PATH"))
+	index := strings.LastIndex(pwd, TanzuCLI)
+	if index < 0 {
+		log.Error(fmt.Sprintf("The current directory path %s does not have %s directory, %s and %s not set in PATH", pwd, TanzuCLI, TanzuCLIHackToolsBin, TanzuCLIBin))
+	} else {
+		hackBin := pwd[:index] + TanzuCLIHackToolsBin
+		cliBin := pwd[:index] + TanzuCLIBin
+		os.Setenv("PATH", hackBin+":"+cliBin+":"+os.Getenv("PATH"))
+	}
```
Comment (anujc25) [inline test/e2e/framework/framework.go:110]: Configuring `PATH` variable as part of the test means, we are assuming how binaries are organized. What if I am running these tests in some other environment where I have already configured all required `PATH` variables?

Is it possible to remove this logic from here and configure the `PATH` variable as part of `Makefile` from where we are invoking `make e2e-cli-core` target?

File: test/e2e/context/context_k8s_lifecycle_test.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+		It("use context command with incorrect context as input", func() {
+			err := tf.ContextCmd.UseContext(framework.RandomString(4))
+			Expect(err).ToNot(BeNil())
+		})
+		// Test case: test 'tanzu context list' command, should list all contexts created
+		It("list context should have all added contexts", func() {
+			list := GetAvailableContexts(tf, contextNames)
+			Expect(len(list)).To(Equal(len(contextNames)), "list context should have all contexts added in previous tests")
+		})
+		// Test case: test 'tanzu context delete' command, make sure to delete all context's created in previous test cases
```
Comment (anujc25) [inline test/e2e/context/context_k8s_lifecycle_test.go:82]: Let's add one negative test case for `tanzu context delete` where the context specified doesn't exist.

**Review by anujc25 — COMMENTED** [review]
Added a few minor comments. Overall this PR looks good.

File: test/e2e/framework/context_create_operations.go (hunk 2 above)
Comment (chandrareddyp) [reply test/e2e/framework/context_create_operations.go:17]: its not working without --staging option.

❯ tz context create --name tmc13 --endpoint unstable.tmc-dev.cloud.vmware.com            
✖  Failed to obtain access token. Please provide valid VMware Cloud Services API-token -- {"metadata":null,"errorCode":null,"statusCode":400,"requestId":"2ac7e0fcb9594e5c","message":"invalid_grant: Invalid refresh token: xxxx...cbamo","cspErrorCode":"540.120-340.800","moduleCode":540,"traceId":"95c9b01ad3455ddd2dd558a366a6cbbe"} 
❯ tz context create --name tmc13 --endpoint unstable.tmc-dev.cloud.vmware.com --staging

✔  successfully created a TMC context 
ℹ  Checking for required plugins... 
!  unable to automatically sync the plugins from target context. Please run 'tanzu plugin sync' command to sync plugins manually 
❯

File: test/e2e/framework/cluster_kind.go [hunk 5]
```diff
@@ -56,14 +63,48 @@ func (kc *kindCluster) DeleteCluster(name string, args []string) (output string,
...
+	}
+	return "", errors.New("the '" + path + "' kubeconfig file does not have context '" + ctx + "' details")
+}
+
+func (kc *kindCluster) GetClusterContext(kindClusterName string) string {
+	return "kind-" + kindClusterName
+}
+
+func (kc *kindCluster) GetKubeconfigPath() string {
+	return os.Getenv("HOME") + "/.kube/config"
```
[the remark this answers is not in the document]
Comment (chandrareddyp) [reply test/e2e/framework/cluster_kind.go]: thank you

File: test/e2e/framework/framework.go (hunk 3 above)
Comment (chandrareddyp) [reply test/e2e/framework/framework.go]: updated in makefile

File: test/e2e/context/context_k8s_lifecycle_test.go (hunk 4 above)
Comment (chandrareddyp) [reply test/e2e/context/context_k8s_lifecycle_test.go:82]: added

**Review by anujc25 — APPROVED** [review]
Looks good.

**Pull request merged**